boot/zephyr/decompression.c

```diff
@@ -126,18 +127,23 @@ int bootutil_img_hash_decompress(struct enc_key_data *enc_state, int image_index
         goto finish_without_clean;
     }
 
-    compression = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_LZMA);
+    compression_lzma = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_LZMA);
+    compression_arm_thumb = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_ARM_THUMB);
 
-    if (compression == NULL || compression->init == NULL || compression->deinit == NULL ||
-        compression->decompress_bytes_needed == NULL || compression->decompress == NULL) {
+    if (compression_lzma == NULL || compression_lzma->init == NULL ||
```

Comment:
This basically begs for "is object valid" kind of function.

Reply:
Fixed